### PR TITLE
Unbalanced join detector

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

If the round is unbalanced in favor of marines and the xenos still have a lot of larvas (more than 3), the game will warn player when they try to join as marine, asking them to consider joining as a xenos instead

## Why It's Good For The Game

Capture

## Changelog
:cl:
add: Popup when joining as a marine and xenos are losing
/:cl:

### CODE REVIEW

**Comment by XSlayer300** [comment]
How about the reverse? Just in case that they're losing groundside?

**Comment by BraveMole** [comment]
I never saw a lack of marines

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -255,9 +263,28 @@
...
 				return
 			var/datum/job/job_datum = locate(href_list["job_selected"])
+			if(!isxenosjob(job_datum) && (SSmonitor.current_state == XENOS_LOSING || SSmonitor.current_state == XENOS_DELAYING))
+				var/datum/job/xeno_job = SSjob.GetJobType(/datum/job/xenomorph)
+				if((xeno_job.total_positions-xeno_job.current_positions) >= 3)
+					var/datum/browser/popup = new(src, "xenosunbalanced", "Warning: the game is unbalanced", 450, 250)
+					popup.set_content(reconsider_job_panel())
+					popup.open(FALSE)
+					saved_job = job_datum
+					return
```
Comment (SplinterGP) [inline code/modules/mob/new_player/new_player.dm:273]: Just use tgui_alert() instead.

**Review by SplinterGP — APPROVED** [review]

**Comment by SplinterGP** [comment]
MISSCLICK AGH DINDT MEAN TO APPROVE

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -31,6 +33,12 @@
 	new_character = null
 	return ..()
 
+/mob/new_player/proc/reconsider_job_panel()
+	var/output = "<div align='center'>"
+	output += "<p>There is a lack of xenos players on this round, unbalanced rounds are unfun for everyone.</i></p>"
+	output += "<p><b>Are you sure you want to play as a marine?</b></p>"
+	output += "<br><p><a href='byond://?src=[REF(src)];lobby_choice=continue_join'>Yes</A> | <a href='byond://?src=[REF(src)];lobby_choice=reconsider'> No </A></p>"
+	return output
```
Comment (SplinterGP) [inline code/modules/mob/new_player/new_player.dm:41]: ditto use tgui_alert() instead if possible

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/mob/new_player/new_player.dm (hunk 2 above)
Comment (BraveMole) [reply code/modules/mob/new_player/new_player.dm:41]: TOO LATE YOU APPROVED I WONT DO NOTHING AH

Comment (SplinterGP) [reply code/modules/mob/new_player/new_player.dm:41]: > TOO LATE YOU APPROVED I WONT DO NOTHING AH

SUFFER DO NOTHING NO MERGE

**Comment by BraveMole** [comment]
Waiting for tivi's ui to get merge

**Comment by SplinterGP** [comment]
> Waiting for tivi's ui to get merge

(y)

**Review by TiviPlus — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-38). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**